### PR TITLE
Rearch: logging interface

### PR DESCRIPTION
**Pull request by rihp**
### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by dwrtz** [comment]
If I wanted to add additional metadata to a log message, what would be the recommend way to do that? I suppose it could be part of the message string. I'm thinking information like timestamp, user_id, agent_id.

Or perhaps an optional meta field in the LogMessage class. Something like this?
```python
class LogMessage:
  title: str
  message: str
  format: LogFormat
  meta: Optional[dict]
```

Just thinking out loud here

**Comment by dwrtz** [comment]
Otherwise this looks great. Being able to have a logger instance per-agent will be a huge win

**Comment by anonhostpi** [comment]
This is good progress. Keep up the good work!

**Comment by rihp** [comment]
Made the changes suggested

**Comment by rihp** [comment]
We should consider adding a handler for syncing logs

File: autogpt/core/logging/base.py [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+    WARN = 2
+    INFO = 3
+    DEBUG = 4
+    SILLY = 5
 
 
 class Logger(abc.ABC):
-    pass
+    @abc.abstractmethod
+    def __init__(self, *arg, **kwargs):
```
Comment (ntindle) [inline autogpt/core/logging/base.py:34]: Require agent ID here

**Pull request merged**